ocs_ci/ocs/cluster.py

```diff
@@ -1209,6 +1209,37 @@ def check_osd_tree_1az_aws(osd_tree, number_of_osds):
     return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
 
 
+def check_ceph_osd_tree():
+    """
+    Checks whether an OSD tree is created/modified correctly.
+    It is a summary of the previous functions: 'check_osd_tree_1az_vmware',
+    'check_osd_tree_3az_aws', 'check_osd_tree_1az_aws'.
+
+    Returns:
+         Boolean: True, if the ceph osd tree is formed correctly. Else False
+
+    """
+    osd_pods = pod.get_osd_pods()
+    # 'ceph osd tree' should show the new osds under right nodes/hosts
+    #  Verification is different for 3 AZ and 1 AZ configs
+    ct_pod = pod.get_ceph_tools_pod()
+    tree_output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph osd tree')
+    if config.ENV_DATA['platform'].lower() == constants.VSPHERE_PLATFORM:
+        return check_osd_tree_1az_vmware(tree_output, len(osd_pods))
+
```

Comment:
@yitzhak12 are we confirming that the following:

a) the new OSD is in UP state in ceph osd tree (AFAIU check_osd_tree_1az_vmware only checks the OSD is there or not)
b) Do we verify that the new deviceset is added to the same rack ( @prsurve after node replacement, does the host, i.e. ocs-deviceset-xxx name change? Is a new OSD with ocs-deviceset-y-xxx is created ?)

Reply:
a) I added another check to see that the OSD's are in 'up' state. 
b) After node replacement, I don't see any change in the ocs-deviceset-xxx names at the osd tree. I checked both in AWS and vSphere. Maybe we just need to check that the new OSD and the new host added to the same rack.

Reply:
Hi @yitzhak12 thanks for adding the check for "UP status"

One query - for device replacement, the **ocs-deviceset-<0/1/2>-xxx** name changes, how come it doesn't change for node replacement ?Isn't the end result same (OSD is replaced ?) . Sorry I was assuming that the name would change

e.g. from https://bugzilla.redhat.com/show_bug.cgi?id=1841461

```

After the OSD has been replaced:
================================

ID  CLASS WEIGHT  TYPE NAME                            STATUS REWEIGHT PRI-AFF 
 -1       1.49698 root default                                                 
-12       0.49899     rack rack0                                               
-11       0.49899         host ocs-deviceset-0-0-tc8lq                         
  2   hdd 0.49899             osd.2                        up  1.00000 1.00000 
 -8       0.49899     rack rack1                                               
 -7       0.49899         host ocs-deviceset-1-0-psv6f                         
  1   hdd 0.49899             osd.1                        up  1.00000 1.00000 
 -4       0.49899     rack rack2                                               
 -3             0         host ocs-deviceset-2-0-24bs7                         
-15       0.49899         host ocs-deviceset-2-0-bnn6p                         
  0   hdd 0.49899             osd.0                        up  1.00000 1.00000 
```

Reply:
iirc, The ocs-deviceset name will remain in case of Dynamically provsioned cluster, but we do expect changes in case of LSO setup.
Previously we hit the [Bug 1840539](https://bugzilla.redhat.com/show_bug.cgi?id=1840539) on LSO setup

Reply:
In Dynamic cluster, ocs-deviceset name will remain the same but in LSO case host name will be the node name not the pvc name

Reply:
ceph osd tree output from LSO cluster

```ceph osd tree
ID  CLASS WEIGHT  TYPE NAME                            STATUS REWEIGHT PRI-AFF 
 -1       2.72910 root default                                                 
 -4       0.90970     rack rack0                                               
 -3       0.90970         host argo007-ceph-redhat-com                         
  0   ssd 0.90970             osd.0                        up  1.00000 1.00000 
 -8       0.90970     rack rack1                                               
 -7       0.90970         host argo005-ceph-redhat-com                         
  1   ssd 0.90970             osd.1                        up  1.00000 1.00000 
-12       0.90970     rack rack2                                               
-11       0.90970         host argo006-ceph-redhat-com                         
  2   ssd 0.90970             osd.2                        up  1.00000 1.00000 
```

Reply:
Currently, I need to add another test for node replacement for LSO. After I add this test for LSO, I will add the verification steps specificaly for LSO. I think it's not something that need to prevent this PR for merging

Reply:
+1 @yitzhak12